### PR TITLE
Rule that suggest using auto-property

### LINKED ISSUE #596: Suggest usage of member val Foo when member self.Foo is unnecessary
Example: https://github.com/fsprojects/fantomas/commit/fe4e11f887923082ca52b2fbe6711461de132f9d

### PR DESCRIPTION
**Pull request by webwarrior-ws**
Add rule that suggests usage of auto-property (`member val Foo`) when `member self.Foo` is unnecessary.

Fixes #596.

### CODE REVIEW

File: src/FSharpLint.Core/Rules/Conventions/SuggestUseAutoProperty.fs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+open System
+open FSharpLint.Framework
+open FSharpLint.Framework.Suggestion
+open FSharp.Compiler.Syntax
+open FSharpLint.Framework.Ast
+open FSharpLint.Framework.Rules
+
+let rec private isImmutableValueExpression (args: AstNodeRuleParams) (expression: SynExpr) =
+    match expression with
+    | SynExpr.Const (constant, range) -> true
```
Comment (knocte) [inline src/FSharpLint.Core/Rules/Conventions/SuggestUseAutoProperty.fs:12]: @webwarrior-ws please add _ prefix to unused vars

File: src/FSharpLint.Core/Text.resx [hunk 2]
```diff
@@ -348,4 +348,7 @@
   <data name="RulesAvoidSinglePipeOperator" xml:space="preserve">
     <value>Can use normal function call (instead of pipe operator) because there is no invocation chain (single invocation only).</value>
   </data>
-</root>
+  <data name="RulesSuggestUseAutoProperty" xml:space="preserve">
+    <value>Consider giving auto-properties via the "val" keyword</value>
```
Comment (knocte) [inline src/FSharpLint.Core/Text.resx:352]: @webwarrior-ws please be consistent with other messages above, which add a period at the end

File: src/FSharpLint.Core/Text.resx [hunk 3]
```diff
@@ -348,4 +348,7 @@
...
+  </data>
+</root>
```
Comment (knocte) [inline src/FSharpLint.Core/Text.resx:354]: @webwarrior-ws please add EOL at EOF

File: src/FSharpLint.Core/Rules/Conventions/SuggestUseAutoProperty.fs (hunk 1 above)
Comment (webwarrior-ws) [reply src/FSharpLint.Core/Rules/Conventions/SuggestUseAutoProperty.fs]: done

File: src/FSharpLint.Core/Text.resx (hunk 2 above)
Comment (webwarrior-ws) [reply src/FSharpLint.Core/Text.resx]: done

**Comment by knocte** [comment]
@webwarrior-ws can you rebase this please?

**Comment by knocte** [comment]
@webwarrior-ws also update the *Introduced in...* text.

**Comment by webwarrior-ws** [comment]
@knocte rebased

File: src/FSharpLint.Core/Rules/Conventions/SuggestUseAutoProperty.fs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+              Message = Resources.GetString "RulesSuggestUseAutoProperty"
+              SuggestedFix = Some suggestedFix
+              TypeChecks = List.Empty }
+            |> Array.singleton
+        | _ -> Array.empty
+    | _ -> Array.empty
+
+let rule =
+    { Name = "SuggestUseAutoProperty"
+      Identifier = Identifiers.FavourConsistentThis
```
Comment (knocte) [inline src/FSharpLint.Core/Rules/Conventions/SuggestUseAutoProperty.fs:95]: @webwarrior-ws this is a typo here, please fix

Comment (webwarrior-ws) [reply src/FSharpLint.Core/Rules/Conventions/SuggestUseAutoProperty.fs]: Fixed

**Pull request merged**